inlong-sort/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/table/PulsarDynamicTableSource.java

```diff
@@ -233,35 +234,61 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext context) {
                 valueDeserialization,
                 producedTypeInfo);
         final ClientConfigurationData clientConfigurationData = PulsarClientUtils.newClientConf(serviceUrl, properties);
-        FlinkPulsarSource<RowData> source = new FlinkPulsarSource<>(
-                adminUrl,
-                clientConfigurationData,
-                deserializationSchema,
-                properties
-        );
-
-        if (watermarkStrategy != null) {
-            source.assignTimestampsAndWatermarks(watermarkStrategy);
-        }
-
-        switch (startupOptions.startupMode) {
-            case EARLIEST:
-                source.setStartFromEarliest();
-                break;
-            case LATEST:
-                source.setStartFromLatest();
-                break;
-            case SPECIFIC_OFFSETS:
-                source.setStartFromSpecificOffsets(startupOptions.specificOffsets);
-                break;
-            case EXTERNAL_SUBSCRIPTION:
-                MessageId subscriptionPosition = MessageId.latest;
-                if (CONNECTOR_STARTUP_MODE_VALUE_EARLIEST.equals(startupOptions.externalSubStartOffset)) {
-                    subscriptionPosition = MessageId.earliest;
-                }
-                source.setStartFromSubscription(startupOptions.externalSubscriptionName, subscriptionPosition);
+        if (adminUrl != null) {
+            FlinkPulsarSource<RowData> source = new FlinkPulsarSource<>(
+                    adminUrl,
+                    clientConfigurationData,
+                    deserializationSchema,
+                    properties
+            );
+
+            if (watermarkStrategy != null) {
+                source.assignTimestampsAndWatermarks(watermarkStrategy);
+            }
+
+            switch (startupOptions.startupMode) {
+                case EARLIEST:
+                    source.setStartFromEarliest();
+                    break;
+                case LATEST:
+                    source.setStartFromLatest();
+                    break;
+                case SPECIFIC_OFFSETS:
+                    source.setStartFromSpecificOffsets(startupOptions.specificOffsets);
+                    break;
+                case EXTERNAL_SUBSCRIPTION:
+                    MessageId subscriptionPosition = MessageId.latest;
+                    if (CONNECTOR_STARTUP_MODE_VALUE_EARLIEST.equals(startupOptions.externalSubStartOffset)) {
+                        subscriptionPosition = MessageId.earliest;
+                    }
+                    source.setStartFromSubscription(startupOptions.externalSubscriptionName, subscriptionPosition);
+            }
+            return SourceFunctionProvider.of(source, false);
+        } else {
+            FlinkTDMQSource<RowData> source = new FlinkTDMQSource<>(
```

Comment:
when admin - url is empty doesn't mean the user want to create tdmqSource

Reply:
ok, add an option to support tmdq option